packages/server/src/validators/HeaderValidator.ts

```diff
@@ -0,0 +1,107 @@
+import {TUS_VERSION, TUS_RESUMABLE} from '../constants'
+import {Metadata} from '../models'
+
+type validator = (value?: string) => boolean
+
+export const validators = new Map<string, validator>([
+  [
+    // The Upload-Offset request and response header indicates a byte offset within a resource.
+    // The value MUST be a non-negative integer.
+    'upload-offset',
+    function (value) {
+      return Number.parseInt(value as string, 10) >= 0
+    },
+  ],
+  [
+    // The Upload-Length request and response header indicates the size of the entire upload in bytes.
+    // The value MUST be a non-negative integer.
+    'upload-length',
+    function (value) {
+      return Number.parseInt(value as string, 10) > 0
+    },
+  ],
+  [
+    // The Upload-Defer-Length request and response header indicates that the size of the upload
+    // is not known currently and will be transferred later.
+    // Its value MUST be 1. If the length of an upload is not deferred, this header MUST be omitted.
+    'upload-defer-length',
+    function (value) {
+      return Number.parseInt(value as string, 10) === 1
```

Comment:
What about a simple string comparison?
```suggestion
      return value === '1'
```